src/DurableTask.Netherite/StorageLayer/Faster/AzureBlobs/AzureStorageDevice.cs

```diff
@@ -448,13 +451,15 @@ async Task WritePortionToBlobAsync(UnmanagedMemoryStream stream, BlobEntry blobE
         {
             using (stream)
             {
+                var position = destinationAddress + offset;
                 long originalStreamPosition = stream.Position;
                 await this.BlobManager.PerformWithRetriesAsync(
                     BlobManager.AsynchronousStorageWriteMaxConcurrency,
                     true,
                     "PageBlobClient.UploadPagesAsync",
                     "WriteToDevice",
-                    $"id={id} length={length} destinationAddress={destinationAddress + offset}",
+                    position,
+                    $"id={id} position={position} length={length}",
```

Comment:
Help me understand: why is position here `destinationAddress +offset` and not `sourceAddress + offset`? Perhaps I misunderstand what `source` and `destination` correspond to in this case?

Reply:
If "the page blob position [...] is the destination", then is it a problem that it we currently have `var position = destinationAddress + offset` instead of just `destination`?